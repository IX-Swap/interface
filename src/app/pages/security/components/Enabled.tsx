import React from 'react'
import { Grid, Typography } from '@mui/material'
import CheckCircleIcon from '@mui/icons-material/CheckCircle'
import { green } from '@mui/material/colors'
import { VSpacer } from 'components/VSpacer'

export const Enabled = () => {
  return (
    <Grid container direction='column' spacing={3} alignItems='center'>
      <Grid item>
        <CheckCircleIcon
          fontSize='large'
          style={{ color: green[500], width: 120, height: 120 }}
        />
        <VSpacer size='small' />
      </Grid>
      <Grid item>
        <Typography align='center' variant='h5'>
          2FA Enabled
        </Typography>
      </Grid>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          Your account is now protected by Two-Factor Authentication
        </Typography>
      </Grid>
    </Grid>
  )
}
