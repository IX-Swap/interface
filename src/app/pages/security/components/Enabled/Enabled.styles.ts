import { makeStyles } from '@mui/styles'

export const useStyles = makeStyles(theme => ({
  container: {
    padding: theme.spacing(0, 0, 8),

    [theme.breakpoints.down('md')]: {
      padding: 0
    }
  },
  text: {
    margin: theme.spacing(2, 'auto', 12.25),
    fontSize: 16,
    opacity: 0.8,
    maxWidth: 484,

    [theme.breakpoints.down('md')]: {
      margin: theme.spacing(3, 'auto', 5)
    }
  },
  icon: {
    maxWidth: '100%'
  }
}))
