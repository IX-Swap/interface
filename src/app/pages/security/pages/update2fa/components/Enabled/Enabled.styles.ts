import { makeStyles } from '@mui/styles'

export const useStyles = makeStyles(theme => ({
  text: {
    maxWidth: 310,
    margin: theme.spacing(3, 'auto', 10),
    fontSize: 18
  }
}))
