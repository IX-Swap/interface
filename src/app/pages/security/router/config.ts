import { makeURL } from 'config/appURL'

export const SecurityRoute = {
  landing: makeURL(['app', 'settings']),
  changePassword: makeURL(['app', 'settings', 'changePassword']),
  setup2fa: makeURL(['app', 'settings', 'setup2fa']),
  guide: makeURL(['app', 'settings', 'guide2fa'])
}
