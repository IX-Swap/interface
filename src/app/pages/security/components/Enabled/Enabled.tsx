import React from 'react'
import { Grid, Typography } from '@mui/material'
import Enabled2FaIcon from 'app/pages/security/pages/update2fa/assets/enabled-2fa.png'
import { useStyles } from 'app/pages/security/components/Enabled/Enabled.styles'

export interface EnabledProps {
  update2FA?: boolean
}

export const Enabled = ({ update2FA = false }: EnabledProps) => {
  const classes = useStyles()
  const title = update2FA ? (
    <>
      2FA Has Been Updated
      <br /> Successfully
    </>
  ) : (
    '2FA Enabled'
  )

  return (
    <Grid
      container
      direction='column'
      alignItems='center'
      className={classes.container}
    >
      <Grid item container alignItems={'center'} justifyContent={'center'}>
        <Grid>
          <Typography align='center' variant='h4'>
            {title}
          </Typography>
        </Grid>
      </Grid>
      <Grid item>
        <Typography align='center' variant='body1' className={classes.text}>
          Your account is now protected by Two-Factor Authentication
        </Typography>
      </Grid>
      <Grid item>
        <img
          src={Enabled2FaIcon}
          alt={'Enabled 2FA'}
          className={classes.icon}
        />
      </Grid>
    </Grid>
  )
}
