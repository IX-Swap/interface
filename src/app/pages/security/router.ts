import { InternalRouteProps } from 'types/util'
import { generateAppRouterHook } from 'helpers/generateAppRouterHook'
import { ChangePassword } from './pages/changePassword/ChangePassword'
import { Landing } from './pages/landing/Landing'
import { Setup2fa } from './pages/setup2fa/Setup2fa'
import { makeURL } from 'config/appURL'

export const SecurityRoute = {
  landing: makeURL(['app', 'settings']),
  changePassword: makeURL(['app', 'settings', 'changePassword']),
  setup2fa: makeURL(['app', 'settings', 'setup2fa'])
}

const securityRoutes: InternalRouteProps[] = [
  {
    label: 'Security',
    path: SecurityRoute.landing,
    exact: true,
    component: Landing
  },
  {
    label: 'Change Password',
    path: SecurityRoute.changePassword,
    component: ChangePassword
  },
  {
    label: 'Google Authenticator Setup',
    path: SecurityRoute.setup2fa,
    component: Setup2fa
  }
]

export const useSecurityRouter = generateAppRouterHook<typeof SecurityRoute>(
  SecurityRoute,
  SecurityRoute.landing,
  securityRoutes
)
