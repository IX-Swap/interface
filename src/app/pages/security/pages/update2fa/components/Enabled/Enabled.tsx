import React from 'react'
import { Button, Grid, Typography } from '@mui/material'
import { ReactComponent as CongratsIcon } from 'assets/icons/congrats.svg'
import { history } from 'config/history'
import { SecurityRoute } from 'app/pages/security/router/config'
import { useStyles } from './Enabled.styles'

export const Enabled = () => {
  const classes = useStyles()

  return (
    <Grid container direction='column' alignItems='center'>
      <Grid item container alignItems={'center'} justifyContent={'center'}>
        <Grid>
          <Typography align='center' variant='h4'>
            2-FA Has Been Updated Successfully.
          </Typography>
        </Grid>
        <Grid>
          <CongratsIcon />
        </Grid>
      </Grid>
      <Grid item>
        <Typography align='center' variant='body1' className={classes.text}>
          Your account is now protected by Two-Factor Authentication (2-FA).
        </Typography>
      </Grid>
      <Grid item>
        <Button
          color={'primary'}
          variant={'contained'}
          onClick={() => history.push(SecurityRoute.landing)}
        >
          Ok
        </Button>
      </Grid>
    </Grid>
  )
}
