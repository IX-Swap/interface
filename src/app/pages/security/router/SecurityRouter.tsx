import { ChangePassword } from 'app/pages/security/pages/changePassword/ChangePassword'
import { Landing } from 'app/pages/security/pages/landing/Landing'
import { Setup2fa } from 'app/pages/security/pages/setup2fa/Setup2fa'
import { Setup2FAGuide } from 'app/pages/security/pages/setup2faguide/Setup2FAGuide'
import { Update2fa } from 'app/pages/security/pages/update2fa/Update2fa'
import { SecurityRoute } from 'app/pages/security/router/config'
import { AppRoute } from 'components/AppRoute'
import { privateClassNames } from 'helpers/classnames'
import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { AppContentWrapper } from 'ui/AppContentWrapper'
import { RootContainer } from 'ui/RootContainer'

export const SecurityRouter = () => {
  return (
    <Switch>
      <AppRoute path={SecurityRoute.setup2fa}>
        <RootContainer className={privateClassNames()}>
          <Setup2fa />
        </RootContainer>
      </AppRoute>

      <AppRoute path={SecurityRoute.guide}>
        <RootContainer className={privateClassNames()}>
          <Setup2FAGuide />
        </RootContainer>
      </AppRoute>

      <Route path={SecurityRoute.changePassword}>
        <AppContentWrapper container>
          <ChangePassword />
        </AppContentWrapper>
      </Route>

      <AppRoute path={SecurityRoute.change2fa}>
        <RootContainer className={privateClassNames()}>
          <Update2fa />
        </RootContainer>
      </AppRoute>

      <AppRoute path={SecurityRoute.landing}>
        <AppContentWrapper container>
          <Landing />
        </AppContentWrapper>
      </AppRoute>
    </Switch>
  )
}
