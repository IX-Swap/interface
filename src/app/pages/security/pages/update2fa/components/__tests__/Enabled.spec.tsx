import React from 'react'
import { render } from 'test-utils'
import { Enabled } from 'app/pages/security/pages/update2fa/components/Enabled/Enabled'
import { fireEvent, waitFor } from '@testing-library/dom'
import { history } from 'config/history'
import { SecurityRoute } from 'app/pages/security/router/config'

describe('Enabled', () => {
  it('should match snapshot, invokes history push on button click', async () => {
    const { container, getByText } = render(<Enabled />)
    const button = getByText(/ok/i)

    expect(container).toMatchSnapshot()

    fireEvent.click(button)
    await waitFor(() => {
      expect(history.location.pathname).toEqual(SecurityRoute.landing)
    })
  })
})
