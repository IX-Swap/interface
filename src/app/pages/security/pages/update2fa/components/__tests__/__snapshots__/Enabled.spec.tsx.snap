// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enabled renders 2fa key and image 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-30jy6e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1o62c1b-MuiGrid-root"
    >
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-118amwv-MuiTypography-root"
        >
          2-FA Has Been Updated Successfully.
        </h4>
      </div>
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-text-1 css-6bx0l6-MuiTypography-root"
      >
        Your account is now protected by Two-Factor Authentication (2-FA).
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1i8ofwk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Ok
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Enabled should match snapshot, invokes history push on button click 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-30jy6e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1o62c1b-MuiGrid-root"
    >
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-118amwv-MuiTypography-root"
        >
          2-FA Has Been Updated Successfully.
        </h4>
      </div>
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-text-1 css-6bx0l6-MuiTypography-root"
      >
        Your account is now protected by Two-Factor Authentication (2-FA).
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1i8ofwk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Ok
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
